### PR TITLE
openmolcas: new package at 19.11

### PR DESCRIPTION
**Pull request by codeandkey**

### CODE REVIEW

File: var/spack/repos/builtin/packages/openmolcas/package.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    """OpenMolcas is a quantum chemistry software package.
+       The key feature of OpenMolcas is the multiconfigurational approach to
+       the electronic structure."""
+
+    homepage = "https://gitlab.com/Molcas/OpenMolcas"
+    url      = "https://github.com/Molcas/OpenMolcas/archive/v19.11.tar.gz"
+
+    version('19.11', sha256='8ebd1dcce98fc3f554f96e54e34f1e8ad566c601196ee68153763b6c0a04c7b9')
+
+    depends_on('cmake', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/openmolcas/package.py]: You can remove this, it's already present in the CMakePackage base class

File: var/spack/repos/builtin/packages/openmolcas/package.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    depends_on('hdf5')
+    depends_on('lapack')
+    depends_on('openblas+ilp64')
+    depends_on('python@3.7:', type=('build', 'run'))
+    depends_on('py-pyparsing', type=('build', 'run'))
+    depends_on('py-six', type=('build', 'run'))
+
+    def setup_environment(self, spack_env, run_env):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/openmolcas/package.py]: Need to split into `setup_build_environment` and `setup_run_environment`

File: var/spack/repos/builtin/packages/openmolcas/package.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        spack_env.set('MOLCAS', self.prefix)
+        run_env.set('MOLCAS', self.prefix)
+
+    def cmake_args(self):
+        return [
+            '-DLINALG=OpenBLAS',
+            '-DOPENBLASROOT=%s' % self.spec['openblas'].prefix,
+            '-DPYTHON_EXECUTABLE=%s' % self.spec['python'].command,
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/openmolcas/package.py]:
```suggestion
            '-DPYTHON_EXECUTABLE=%s' % self.spec['python'].command.path,
```
I assume this is necessary, but not sure. `command` is an `Executable` object, `command.path` is a `str`.

File: var/spack/repos/builtin/packages/openmolcas/package.py (hunk 2 above)
Comment (codeandkey) [reply var/spack/repos/builtin/packages/openmolcas/package.py]: Fixed! Hopefully I got the signature right.

File: var/spack/repos/builtin/packages/openmolcas/package.py (hunk 3 above)
Comment (codeandkey) [reply var/spack/repos/builtin/packages/openmolcas/package.py]: Fixed! Executable has a __str__ cast but never hurts to be more verbose.

File: var/spack/repos/builtin/packages/openmolcas/package.py (hunk 1 above)
Comment (codeandkey) [reply var/spack/repos/builtin/packages/openmolcas/package.py]: Fixed!

**Pull request merged**